src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -545,7 +546,7 @@ internal void InitializeConstructorParameters(JsonParameterInfoValues[] jsonPara
             foreach (KeyValuePair<string, JsonPropertyInfo?> kvp in PropertyCache.List)
             {
                 JsonPropertyInfo jsonProperty = kvp.Value!;
```

Comment:
Should we replace this bang with an assert as well?

Reply:
this one is ok because we never actually insert nulls there, it's just weird implementation detail of PropertyCache.List